phd-tests/runner/src/config.rs

```diff
@@ -265,9 +265,18 @@ impl RunOptions {
                 // The Git revision of Crucible we depend on is determined when building
                 // `phd-runner` by the build script, so that the `phd-runner` binary can
                 // be run even after moving it out of the Propolis cargo workspace.
-                let commit = env!("PHD_CRUCIBLE_GIT_REV").parse().context(
-                "PHD_CRUCIBLE_GIT_REV must be set to a valid Git revision by the build script",
-            )?;
+                let commit = match env!("PHD_CRUCIBLE_GIT_REV") {
+                    "CANT_GET_YE_CRUCIBLE_SHA" => anyhow::bail!(
+                        "Crucible upstairs dependency was patched with a \
+                        local path when PHD was compiled, so the \
+                        `--crucible-downstairs-commit auto` option has been \
+                         disabled."
```

Comment:
If we can't or don't want to go the "try to infer where the binary should be based on the path to the checkout" route, perhaps this message (and/or the one in build.rs) should advertise the `--crucible-downstairs-cmd` option so that folks building a local Crucible know they can run PHD with it?

Reply:
Yeah, that's a good idea.

Reply:
It now says:

```
Error: Because Crucible dependency is patched with a local checkout, phd-runner's build script could not determine the Crucible Git SHA, so the `--crucible-downstairs-commit auto` option has been disabled. You can provide a local Crucible binary using `--crucible-downstairs-command`.
```